click/_bashcomplete.py

```diff
@@ -203,15 +237,19 @@ def do_complete(cli, prog_name):
         incomplete = ''
 
     for item in get_choices(cli, prog_name, args, incomplete):
-        echo(item)
+        echo(item[0])
+        if include_descriptions:
+            # ZSH has trouble dealing with empty array parameters when returned from commands, so use a well defined character '_' to indicate no description is present.
```

Comment:
What exactly is the problem here?
Shouldn't echoing an empty string should work?
(but have not looked at the details)

Reply:
Would be simpler to output it as `'%s:%s' % item` - ready for Zsh to consume.
Your approach is more universal, but `%s:%s` could also be parsed for other shells later, and avoids issues with empty descs.